sklearn/decomposition/dict_learning.py

```diff
@@ -717,6 +717,20 @@ def dict_learning_online(X, n_components=2, alpha=1, n_iter=100,
         Number of iterations run. Returned only if `return_n_iter` is
         set to `True`.
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.decomposition import DictionaryLearning
+    >>> X = np.array([[0., 0., 1.],
+    ...     [1.,0.,0.],
+    ...     [2.,2.,2.],
+    ...     [2.,5.,4.]])
+    >>> dico = DictionaryLearning(n_components=2, alpha=1, random_state=1)
+    >>> V = dico.fit(X).components_
+    >>> V
+    array([[-0.2981424 , -0.74535599, -0.59628479],
+           [ 0.57735027,  0.57735027,  0.57735027]])
```

Comment:
Using ellipsis (`...`) and limiting the examples to max 4 digits of floating point precision helps when we compile and test things in different platforms, not all our examples follow this though, [example](https://github.com/scikit-learn/scikit-learn/pull/12124/files). You could also simplify the last two lines and just have `dico.fit(X).components_`, with probably a commend if you like.